pkg/snmp/traps_config.go

```diff
@@ -0,0 +1,94 @@
+package snmp
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/soniah/gosnmp"
+)
+
+// TrapConfig contains configuration for SNMP traps listeners.
+type TrapConfig struct {
+	Host            string
+	Port            uint16
+	Version         string
+	Timeout         int
+	Retries         int
+	Community       string
+	User            string
+	AuthKey         string
+	AuthProtocol    string
+	PrivKey         string
+	PrivProtocol    string
+	ContextEngineID string
+	ContextName     string
```

Comment:
Yes you're right, the traps contain the context info.

I initially copy pasted this code from Thomas' autodiscovery PR. Providing context info in the params there makes sense because the Agent is a sender.

But I realized we're on the receiver end here, so context info is contained in trap packets (as confirmed by the GoSNMP API docs).

Timeout and retries are prob not relevant either in our case so I dropped them.